app/flatpak-builtins-kill.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2018 Red Hat, Inc
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Matthias Clasen <mclasen@redhat.com>
+ */
+
+#include "config.h"
+
+#include <locale.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <signal.h>
+
+#include <glib/gi18n.h>
+
+#include "libglnx/libglnx.h"
+
+#include "flatpak-builtins.h"
+#include "flatpak-instance-private.h"
+
+static GOptionEntry options[] = {
+  { NULL }
+};
+
+static gboolean
+kill_instance (const char *app,
+               GError **error)
+{
+  g_autoptr(GPtrArray) instances = NULL;
+  int j;
+
+  instances = flatpak_instance_get_all ();
+
+  for (j = 0; j < instances->len; j++)
+    {
+      FlatpakInstance *instance = (FlatpakInstance *)g_ptr_array_index (instances, j);
+      if (strcmp (app, flatpak_instance_get_app (instance)) == 0)
+        {
+          pid_t pid = flatpak_instance_get_child_pid (instance);
+          kill (pid, SIGKILL);
+          return TRUE;
+        }
+    }
+
+  return flatpak_fail (error, _("%s is not running."), app);
+}
+
+gboolean
+flatpak_builtin_kill (int            argc,
+                      char         **argv,
+                      GCancellable  *cancellable,
+                      GError       **error)
+{
+  g_autoptr(GOptionContext) context = NULL;
+  const char *app;
+
+  context = g_option_context_new (_("NAME - Stop a running application"));
```

Comment:
Should we not support specifying an instance id too?

Reply:
would be nice. I just wanted to start simple